src/views/MovieListPage.tsx

```diff
@@ -0,0 +1,158 @@
+    setQuery({ ...query, filter: genre === 'All' ? '' : genre });
+  };
+
+  // Sorting
+  const setActiveFilterQuery = (option: IOption) => {
+    setQuery({ ...query, sortBy: option.value });
+  };
+
+  // Movies Select
+  const [activeMovie, setActiveMovie] = useState<IMovieDetails>();
+  const goToActiveMovie = (title: string) => {
+    const activeMovie = moviesList.find((movie) => movie.title === title);
+    setActiveMovie(activeMovie as IMovieDetails);
+    movieDetailsHandler();
+    window.scrollTo({ top: 0, behavior: 'smooth' });
+  };
+  const editMovie = (title: string) => {
+    const activeMovie = moviesList.find((movie) => movie.title === title);
+    setActiveMovie(activeMovie as IMovieDetails);
+    open('MovieModal');
+  };
+  const deleteMovie = (title: string) => {
+    const activeMovie = moviesList.find((movie) => movie.title === title);
+    open(DELETE_MODAL);
+  };
+
+  // Modal
+  const [modalOpen, setModalOpen] = useState<boolean | string>(false);
+  const open = (type: string) => setModalOpen(type);
+  const close = () => setModalOpen(false);
+
+  // Close MovieDetails
+  const closeMovieDetails = () => {
+    setActiveMovie(undefined);
+    movieDetailsHandler();
+  };
+
+  return (
+    <>
+      {activeMovie ? (
+        <MovieDetails
+          movie={activeMovie as IMovieDetails}
+          handleClose={closeMovieDetails}
+        />
+      ) : (
+        <div className={styles.hero}>
+          <div className={styles.heroInner}>
+            <h1 className="hero-title">Find your movie</h1>
+            <SearchForm
+              initialValue={query.search}
+              onSearch={(term) => setQuery({ ...query, search: term })}
+            />
+          </div>
+        </div>
+      )}
+      <div className={styles.movies}>
+        <div className="container">
+          <div className={styles.moviesFilters}>
+            <GenreSelect
+              onSelect={setActiveGenreQuery}
+              genres={genres}
+              activeGenre={activeGenre}
+            />
+            <SortControl onSelected={setActiveFilterQuery} />
+          </div>
+          <div className={styles.moviesResults} data-cy="movies-amount">
+            {moviesList.length ? (
+              `${moviesList.length} movies found`
+            ) : (
+              <div className={styles.moviesEmpty}>
+                <div className="hero-title" data-cy="empty-title">
+                  No movies found
+                </div>
+              </div>
+            )}
+          </div>
+          <div className={styles.moviesCards}>
+            {isLoading
+              ? 'Is loading...'
+              : moviesList.map((movie) => (
+                  <MovieTile
+                    key={movie.id}
+                    movie={movie}
+                    onMovieClick={() => goToActiveMovie(movie.title)}
+                    onMovieDelete={() => deleteMovie(movie.title)}
+                    onMovieEdit={() => editMovie(movie.title)}
```

Comment:
It's better to avoid calling functions with param like this, you can easily call these functions with titles inside MovieTile component.